enclave-tls/src/tls_wrappers/api/tls_wrapper_verify_certificate_extension.c

```diff
@@ -16,24 +16,16 @@ tls_wrapper_err_t tls_wrapper_verify_certificate_extension(tls_wrapper_ctx_t *tl
 {
 	ETLS_DEBUG("tls_wrapper_verify_certificate_extension() called with evidence type: '%s'\n", evidence->type);
 
-	for (unsigned int i = 0; i < registerd_enclave_quote_nums; ++i) {
-		enclave_quote_ctx_t *quote_ctx = enclave_quotes_ctx[i];
-
-		if (strcmp(evidence->type, quote_ctx->opts->type))
-			continue;
-
-		if (!quote_ctx || !(quote_ctx->opts) ||
-		    !(quote_ctx->opts->verify_evidence))
-			return -TLS_WRAPPER_ERR_INVALID;
-
-		//quote_ctx->quote_private = tls_ctx->tls_private;
-		enclave_quote_err_t err =
-			quote_ctx->opts->verify_evidence(quote_ctx, evidence,
-							 hash, hash_len);
-		if (err != ENCLAVE_QUOTE_ERR_NONE) {
-			ETLS_ERR("failed to verify evidence %#x\n", err);
-			return err;
-		}
+	if (!(tls_ctx) || !(tls_ctx->etls_handle) || !(tls_ctx->etls_handle->verifier) || !(tls_ctx->etls_handle->verifier->opts) ||
+	    !(tls_ctx->etls_handle->verifier->opts->verify_evidence))
+		return -TLS_WRAPPER_ERR_INVALID;
+
+	enclave_quote_err_t err =
+		tls_ctx->etls_handle->verifier->opts->verify_evidence(tls_ctx->etls_handle->verifier, evidence,
```

Comment:
Verifier should be determined by evidence type. You are choosing to use a fixed verifier instance to cover all possible retrieved evidence. This is unexpected.

So do you want to implement the dynamic verifier determination in later commit?

Reply:
currently judge evidence type and verifier type. If it's not consistent,  I will return err and message some error info.